### DISCUSSION TITLE
数据量5w的热力图，直接渲染不出来，移除图例也渲染不出来

### DISCUSSION THREAD

**Discussion by nikas1991** [antvis/G2#5051] [Q&A]
官方demo可以跑，实际业务中的数据量达到5w的热力图就渲染不上，已经去除图例

```ts
chart.options({
  type: "view",
  theme: "classic",
  width: 1152,
  height: 780,
  paddingLeft: 60,
  data: vData,
  sort: false,
  legend: false, // 关闭图例
  forceSquare: true,
  scale: {
    color: {
      palette: "turbo",
    },
  },
  axis: {
    y: { labelAutoHide: true, labelAutoRotate: false },
    x: { labelAutoHide: true, labelAutoRotate: false },
  },
  children: [
    {
      type: "cell",
      encode: { x: 'x', y: 'y', color: "z" },
    },
  ],
});
```

**Comment by hustcc** [comment]
可以说一下原始需求嘛，为啥需要用 cell 绘制 5w 的数据量？

**Reply by nikas1991** [reply]
业务需求是需要精确到纳米的单位，当时需求可能是非均匀网格大小，之前项目是用canvas绘制网格线再填充颜色。现在业务需要有图表的交互，数据量在后端会有接近300*300的网格，对应的网格数据就会轻松超过5w

**Reply by hustcc** [reply]
用来显示什么？怎么感觉像是地图中的瓦片？能来一个截图什么的嘛？

**Reply by nikas1991** [reply]
![image](https://github.com/antvis/G2/assets/9650146/fe5656eb-7113-4ca6-94a0-c7186faacd7b)

**Reply by hustcc** [reply]
是不是应该用 heatmap 做，用 cell 做的话确实会很消耗性能的。已经在 PR 中了 https://github.com/antvis/G2/pull/5047 ，欢迎来 CR 代码。

|  ![image](https://github.com/antvis/G2/assets/7856674/79c4f494-825c-432a-8264-6f2a5a3013a6)  |  ![image](https://github.com/antvis/G2/assets/7856674/24f27cd6-ba2b-4f2d-a342-9c70ef1acbf3)  |  ![image](https://github.com/antvis/G2/assets/7856674/07173d4b-9058-4666-8f3c-7873618787b5)  |
|---|---|---|

**Reply by nikas1991** [reply]
因为业务场景就需要每个点的精确度，才会有这么大的量级，用heatmap会丢失精确的特性，现在用d3在做简单渲染

**Reply by hustcc** [reply]
这么大数据不聚合，再这样大小的画布中，其实也只能看到一些分布大概的信息，不过如果确实非要这么做，要那么高性能，只能优化掉数据处理的性能损耗，直接用 canvas，比特是 webgl 性能会很好。

**Comment by peaceps** [comment]
你可以试试把动画关掉 .animate(false)
我从G2v2升上来也碰到这个问题，数量大的时候绝大部分时间都是在做animation
关闭animate之后 9W条数据的point()大概耗时10s 虽然还是比v2的时候慢一倍 但也还可以接受

**Reply by hustcc** [reply]
@peaceps  感谢反馈，最近已经在做性能的优化了，见[这里](https://www.yuque.com/antv/ou292n/fv4rqhytfb9yu5fu)，期望 3 月底能进行一波优化，整体性能能提升约一倍。

**Reply by qtj4js** [reply]
@hustcc 请问这个性能优化已经发稳定版本了么? 是5.1.18么?

**Reply by hustcc** [reply]
使用当前最新的版本，就是有性能优化的版本了，应该是 5.1.18 以后的都可以。
